openfisca_aotearoa/variables/income_tax/working_for_families/working_for_families.py

```diff
@@ -0,0 +1,105 @@
+from openfisca_aotearoa.entities import Titled_Property, Person, Family
+from numpy import clip, floor
+
+class received_income_tested_benefit_as_per_social_security(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = u'Number of Persons classified as receiving an income tested benefit'
+    reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/whole.html#DLM1518484"
+
+class received_parents_allowance_as_per_veterans_support(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = u'Number of Persons classified as receiving a parents allowance'
+    reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/whole.html#DLM1518484"
+
+class received_childrens_pension_as_per_veterans_support(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = u'Number of Persons classified as receiving a parents allowance'
+    reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/whole.html#DLM1518484"
+
+class resident_as_per_income_tax(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = u'Number of Persons classified as meeting residence requirements'
+    reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/DLM1518482.html"
+    # This should really be a forumla based variable covering the full residency criteria.
+
+
+class income_as_per_income_tax(Variable):
+	value_type = float
+	entity = Person
+	definition_period = YEAR
+	label = u'The annual income for a person in family.'
+	reference = "" # TODO Add
+
+class dependants_as_per_income_tax(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u'Number of Persons classified as financially dependant'
+    reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/whole.html#DLM1520883"
+
+    def formula(person, period, parameters):
+        age = person('age', period)
+        return age <= 18 # It's not this simple, this needs to be tweaked to inlcude the edge criteria.
+
+class eligible_for_working_for_families(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = YEAR
+    label = u'Number of Persons classified as eligible for working for families tax credits'    
+    reference = "http://www.legislation.govt.nz/act/public/2007/0097/latest/DLM1518477.html"
+
+    def formula(person, period, parameters):
+        received_tested_benefit     = person('received_income_tested_benefit_as_per_social_security', period)
+        received_parents_allowance  = person('received_parents_allowance_as_per_veterans_support', period)
+        received_childrens_pension  = person('received_childrens_pension_as_per_veterans_support', period)     
+
+        return received_tested_benefit == False and received_parents_allowance == False and received_childrens_pension == False
```

Comment:
Unfortunately this code cannot work because it is not vectorial 😔
Variables always return a vector (an array) of values, so `received_tested_benefit` for example is actually an array of Booleans of the same length as the number of Persons in the current simulation. You might be testing for 1, but this code could be run for 1,000,000!

Thus, `received_tested_benefit == False` will always return `False` because an array is never `== False`… This should probably be transformed into a conjunction of negations (transforming each vector of Booleans so that their `False` become `True`, and creating a new vector that contains `True` only when all of the vectors contain `True` for that Person).
This can be achieved with `not_` and `*`, as documented in the [relevant doc section](https://openfisca.org/doc/coding-the-legislation/25_vectorial_computing.html#boolean-operations).

[More background information on vectorial computing](https://openfisca.org/doc/coding-the-legislation/25_vectorial_computing.html) in OpenFisca.